docs/readmes/bazel/bazel.md

```diff
@@ -0,0 +1,510 @@
+
+ The most important configurations for Magma are:
+
+- Plain, i.e. no special configuration, this is the development version of the targets.
+- `production`, which is the production version of the targets used for releases.
+- `asan`, a configuration with the address sanitizer enabled, used for finding memory corruption bugs.
+- `lsan`, a configuration with the leak sanitizer enabled, used for finding run-time memory leaks.
+
+### Dependencies
+
+To add a dependency to a bazel target the `deps` attribute should be used. The `deps` attribute exists for all library, binary and test rules. Dependencies are specified as a list of strings. There are several distinct types of dependencies:
+
+- [local dependencies](#local-dependencies)
+- [external dependencies](#external-dependencies)
+- [pip dependencies](#pip-dependencies-python).
+
+A simple example that illustrates these is as follows:
+
+```starlark
+py_library(
+  name = "my_lib",
+  srcs = ["my_lib.py"],
+  deps = [
+  	":my_package_dependency",
+  	"//orc8r/magma/common:my_util",
+  	"@aioh2_repo//:aioh2",
+  	requirement("grpcio"),
+  ],
+)
+```
+
+#### Local dependencies
+
+Dependencies on targets in the **same package**, i.e. the same `BUILD.bazel` file, are specified by the target name, prefixed by `:`. For example as `deps = [":my_package_dependency"]`.
+
+Dependencies on targets in **other packages within Magma** are additionally prefixed by `//` and the relative path of the respective package. For example as `deps = ["//orc8r/magma/common:my_util"]`.
+
+#### External dependencies
+
+External dependencies can be added to `$MAGMA_ROOT/bazel/<language>_repositories.bzl`. The target is either defined in the external source (if it is a bazelified dependency) or in the respective build file in `$MAGMA_ROOT/bazel/external` (if it is a dependency that is not bazelified).
+
+In the `deps` attribute of the target that needs the external dependency the name of the external dependency repository is prefixed by `@` and followed by the bazel target. For example as `deps = ["@aioh2_repo//:aioh2"]`.
+
+#### Pip dependencies (Python)
+
+In most cases external Python dependencies can be added as pip dependencies. For this the dependency needs to be added to the `$MAGMA_ROOT/bazel/external/requirements.in` file. If a new dependency is added, the `requirements.txt` file needs to be rebuild by
+
+```bash
+$MAGMA_ROOT/bazel/external $ pip-compile --allow-unsafe --generate-hashes --output-file=requirements.txt requirements.in
+```
+
+> Info: See the `bazel/external/requirements_README.md` for more information regarding the `requirements.in` file.
+
+The pip dependency, e.g. `grpcio`, can now be added to a target by adding a requirement statement, e.g. `deps = ["requirement("grpcio")"]`.
```

Comment:
I would remove the second "e.g."

Reply:
Do we need the " around `requirement`?

Reply:
@wolfseb The `"` are not needed indeed, good catch!